quarkus-workshop-super-heroes/super-heroes/rest-fights/src/test/java/io/quarkus/workshop/superheroes/fight/FightResourceConsumerTest.java

```diff
@@ -0,0 +1,124 @@
+    port = "8093"
+)
+public class FightResourceConsumerTest {
+
+    private static final String HERO_API_BASE_URI = "/api/heroes";
+    private static final String HERO_RANDOM_URI = HERO_API_BASE_URI + "/random";
+
+    @Pact(consumer = "rest-fights")
+    public V4Pact randomHeroFoundPact(PactDslWithProvider builder) {
+        return builder
+            .uponReceiving("A request for a random hero")
+            .path(HERO_RANDOM_URI)
+            .method(HttpMethod.GET)
+            .headers(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON)
+            .willRespondWith()
+            .status(Response.Status.OK.getStatusCode())
+            .headers(Map.of(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON))
+            .body(newJsonBody(body ->
+                    body
+                        .stringType("name", DefaultTestHero.DEFAULT_HERO_NAME)
+                        .integerType("level", DefaultTestHero.DEFAULT_HERO_LEVEL)
+                        .stringType("picture", DefaultTestHero.DEFAULT_HERO_PICTURE)
+                ).build()
+            )
+            .toPact(V4Pact.class);
+    }
+    // end::skeleton[]
+
+    // tag::randomHeroFound[]
+    @Test
+    @PactTestFor(pactMethod = "randomHeroFoundPact")
+    void randomHeroFound() {
+        Fighters fighters = given()
+            .when()
+            .get("/api/fights/randomfighters")
+            .then()
+            .statusCode(OK.getStatusCode())
+            .contentType(APPLICATION_JSON)
+            .extract()
+            .as(Fighters.class);
+
+        Hero hero = fighters.hero;
+        assertEquals(hero.name, DefaultTestHero.DEFAULT_HERO_NAME);
+        assertEquals(hero.picture, DefaultTestHero.DEFAULT_HERO_PICTURE);
+        assertEquals(hero.level, DefaultTestHero.DEFAULT_HERO_LEVEL);
+
+        // We're really trying to test the fighter, so we want to make sure it still passes back
+        // a villain
+        Villain villain = fighters.villain;
+        assertEquals(villain.name, DefaultTestVillain.DEFAULT_VILLAIN_NAME);
+        assertEquals(villain.picture, DefaultTestVillain.DEFAULT_VILLAIN_PICTURE);
+        assertEquals(villain.level, DefaultTestVillain.DEFAULT_VILLAIN_LEVEL);
+    }
+    // end::randomHeroFound[]
+
+    // tag::randomHeroNotFound[]
+    @Pact(consumer = "rest-fights")
+    public V4Pact randomHeroNotFoundPact(PactDslWithProvider builder) {
+        return builder
+            .given("No random hero found")
+            .uponReceiving("A request for a random hero")
+            .path(HERO_RANDOM_URI)
+            .method(HttpMethod.GET)
+            .headers(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON)
+            .willRespondWith()
+            .status(Status.NOT_FOUND.getStatusCode())
+            .toPact(V4Pact.class);
+    }
+
+    @Test
+    @PactTestFor(pactMethod = "randomHeroNotFoundPact")
+    void shouldGetRandomFighters() {
+        given()
```

Comment:
Should this match what you had above?

![image](https://github.com/quarkusio/quarkus-workshops/assets/363447/efa73aac-5cb9-4c05-a5c6-3c29ad8c8871)

Reply:
Done; and I used an includes, so I only need to change it in one place.